src/java/htsjdk/samtools/SRAFileReader.java

```diff
@@ -0,0 +1,275 @@
+
+    @Override
+    public void close() { }
+
+    @Override
+    public ValidationStringency getValidationStringency() {
+        return validationStringency;
+    }
+
+
+    /** INDEXING */
+
+
+    /**
+     * Returns true if the supported index is browseable, meaning the bins in it can be traversed
+     * and chunk data inspected and retrieved.
+     *
+     * @return True if the index supports the BrowseableBAMIndex interface.  False otherwise.
+     */
+    @Override
+    public boolean hasBrowseableIndex() {
+        return true;
+    }
+
+    /**
+     * Gets an index tagged with the BrowseableBAMIndex interface.  Throws an exception if no such
+     * index is available.
+     *
+     * @return An index with a browseable interface, if possible.
+     * @throws SAMException if no such index is available.
+     */
+    @Override
+    public BrowseableBAMIndex getBrowseableIndex() {
+        return index;
+    }
+
+    /**
+     * Iterate through the given chunks in the file.
+     *
+     * @param chunks List of chunks for which to retrieve data.
+     * @return An iterator over the given chunks.
+     */
+    @Override
+    public SAMRecordIterator iterator(final SAMFileSpan chunks) {
+        CloseableIterator<SAMRecord> it = getIterator(chunks);
+        if (it == null) {
+            return null;
+        }
+        return (SAMRecordIterator) it;
+    }
+
+    /** ReaderImplementation */
+    @Override
+    void enableFileSource(final SamReader reader, final boolean enabled) {
+        log.info("enableFileSource is not supported");
+    }
+
+    @Override
+    void enableIndexCaching(final boolean enabled) {
+        log.info("enableIndexCaching is not supported");
+    }
+
+    @Override
+    void enableIndexMemoryMapping(final boolean enabled) {
+        log.info("enableIndexMemoryMapping is not supported");
+    }
+
+    @Override
+    void enableCrcChecking(final boolean enabled) {
+        log.info("enableCrcChecking is not supported");
+    }
+
+    @Override
+    void setSAMRecordFactory(final SAMRecordFactory factory) {
+        log.info("setSAMRecordFactory is not supported");
+    }
+
+    @Override
+    void setValidationStringency(final ValidationStringency validationStringency) {
+        this.validationStringency = validationStringency;
+    }
+
+    protected SRAIterator.RecordRangleInfo getRecordsRangeInfo() {
+        return recordRangeInfo;
+    }
+
+    private SAMFileHeader loadSamHeader() throws ErrorMsg {
+        if (run == null) {
+            throw new RuntimeException("Cannot load SAMFileHeader - SRA run is uninitialized");
+        }
+
+        String runName = run.getName();
+
+        SAMFileHeader header = new SAMFileHeader();
+        header.setSortOrder(SAMFileHeader.SortOrder.coordinate);
+
+        ReadGroupIterator itRg = run.getReadGroups();
+        while (itRg.nextReadGroup()) {
+            String rgName = itRg.getName();
+            if (rgName.equals(""))
+                rgName = "_";
```

Comment:
Nameless read groups are allowed? Should this be an error? Giving these groups the same name (`"_"`) could cause issues with code that uses `SAMReadGroupRecord`s as keys in a `Map`